src/internal_rep.rs

```diff
@@ -644,13 +745,34 @@ impl<'a> ClassList<'a> {
     }
 
     pub fn is_perm(&self, perm: &str) -> bool {
+        if self.perm_sets.get(perm).is_some() {
+            return true;
+        }
         for class in self.classes.values() {
             if class.contains_perm(perm) {
                 return true;
             }
         }
         false
     }
+
+    pub fn insert_perm_set(&mut self, set_name: &str, perms: Vec<String>) {
+        self.perm_sets.insert(set_name.to_string(), perms);
+    }
+
+    pub fn expand_perm_list(&self, perms: Vec<&HLLString>) -> Vec<HLLString> {
+        let mut ret = Vec::new();
+        for p in perms {
+            if let Some(pset) = self.perm_sets.get(&p.to_string()) {
+                let pset_strings: Vec<HLLString> =
+                    pset.iter().map(|s| HLLString::from(s.as_str())).collect();
```

Comment:
Not sure `.as_str()` is required.

Reply:
There may be other solutions, but just removing the as_str() call gives:

 the trait `From<&String>` is not implemented for `HLLString`

Reply:
Right. Another way to do it would be to have another `From<HLLString>` impl for this, but it's OK for this PR.